kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/int-simplification.k

```diff
@@ -211,36 +213,36 @@ module INT-SIMPLIFICATION-COMMON
     rule [upInt-lt-true]:
       ((X up/Int Y) *Int Y) <Int Z => true
       requires X +Int Y <=Int Z andBool 0 <Int Y
-      [simplification]
+      [simplification, preserves-definedness]
 
     rule [upInt-lt-false]:
       ((X up/Int Y) *Int Y) <Int Z => false
       requires Z <=Int X andBool 0 <Int Y
-      [simplification]
+      [simplification, preserves-definedness]
 
     rule [upInt-refl-leq]:
       ((X up/Int Y) *Int Y) <=Int X => X modInt Y ==Int 0
       requires 0 <Int Y
-      [simplification]
+      [simplification, preserves-definedness]
 
     rule [upInt-refl-gt]:
       X <Int ((X up/Int Y) *Int Y) => X modInt Y =/=Int 0
       requires 0 <Int Y
-      [simplification]
+      [simplification, preserves-definedness]
 
     rule [upInt-refl-geq]:
       X <=Int ((X up/Int Y) *Int Y) => true
       requires 0 <Int Y
-      [simplification]
+      [simplification, preserves-definedness]
 
     rule [upInt-ref-eq]:
       X ==Int ((X up/Int Y) *Int Y) => X modInt Y ==Int 0
       requires 0 <Int Y
-      [simplification, comm]
+      [simplification, comm, preserves-definedness]
 
     rule [upInt-refl-neq]:
```

Comment:
It seems these are all duplications of the rules in `evm-int-simplification.k`....

Reply:
Correct, I was wondering about that already. Might try to remove the `evm-int-simplifications` file altogether and measure the effect